src/components/Articles.jsx

```diff
@@ -1,34 +1,46 @@
-import {useParams} from 'react-router-dom';
+import {useParams,Link} from 'react-router-dom';
 import {useState, useEffect} from 'react';
 import axios from 'axios';
+import Pagination from './Pagination';
 
 function Articles() {
     const params = useParams();
     console.log(params);
     const [articles, setArticles] = useState([]);
+    const [isLoading , setIsLoading] = useState(false);
+    const [page, setPage] = useState(1);
+    const [limit, setLimit] = useState(10);
     //create base url
     useEffect(() => {
-        axios.get('https://news-api-9k2x.onrender.com/api/articles', {params: {topic: params.topic}})
+        setIsLoading(true);
+        axios.get(`https://news-api-9k2x.onrender.com/api/articles?p=${page}&limit=${limit}`, {params: {topic: params.topic}})
         .then(({data}) => {
+            setIsLoading(false);
             setArticles(data.articles);
+        }).catch(err => {
+            setIsLoading(false);
+            console.log('err')
```

Comment:
consider making an error state to handle errors and display relevant messages to the user